frontend/src/features/user-settings/user-settings.tsx

```diff
@@ -0,0 +1,165 @@
+import {
+  Divider,
+  Stack,
+  Text,
+  UnstyledButton,
+  Group,
+  rem,
+  Title,
+  Flex,
+  Breadcrumbs,
+  Anchor,
+} from '@mantine/core'
+import { IconChevronRight, IconArrowLeft } from '@tabler/icons-react'
+import { VisibilitySettings } from '../visibility-settings'
+import { useState } from 'react'
+import { useAppSelector } from '../../redux'
+
+function UserSettings() {
+  type SettingsView =
+    | 'main'
+    | 'visibility'
+    | 'visibility-all'
+    | 'visibility-tags'
+
+  const [currentView, setCurrentView] = useState<SettingsView>('main')
+
+  const hasTags = useAppSelector(
+    (state) => !!Object.keys(state.tableData.metadata.tags).length
+  )
+
+  const BreadcrumbBar = () => {
+    const crumbs =
+      currentView === 'main'
+        ? [{ label: 'User settings' as const }]
+        : currentView === 'visibility'
+          ? [
+              { label: 'User settings' as const, to: 'main' as SettingsView },
+              { label: 'Visibility' as const },
+            ]
+          : currentView === 'visibility-all'
+            ? [
+                { label: 'User settings' as const, to: 'main' as SettingsView },
+                {
+                  label: 'Visibility' as const,
+                  to: 'visibility' as SettingsView,
+                },
+                { label: 'By variable' as const },
+              ]
+            : [
+                { label: 'User settings' as const, to: 'main' as SettingsView },
+                {
+                  label: 'Visibility' as const,
+                  to: 'visibility' as SettingsView,
+                },
+                { label: 'By tag' as const },
+              ]
+
+    return (
+      <Breadcrumbs separator="›">
+        {crumbs.map((c, i) => {
+          const isLast = i === crumbs.length - 1
+          if (isLast || !('to' in c) || !c.to) {
+            return (
+              <Text key={`${c.label}-${i}`} c={isLast ? undefined : 'dimmed'}>
+                {c.label}
+              </Text>
+            )
+          }
+          return (
+            <Anchor
+              key={`${c.label}-${i}`}
+              href="#"
+              c="black"
+              fw={600}
+              onClick={(e) => {
+                e.preventDefault()
+                setCurrentView(c.to!)
+              }}
+            >
+              {c.label}
+            </Anchor>
+          )
+        })}
+      </Breadcrumbs>
+    )
+  }
+
+  if (currentView === 'visibility-all') {
```

Comment:
This could be a good use case for `switch`. There's also some similarity as a base (`Stack`, `BreadcrumbBar`), maybe check if the contents can be conditionally rendered instead?